ext/regex.go

```diff
@@ -330,3 +342,80 @@ func extractAll(target, regexStr ref.Val) ref.Val {
 	}
 	return types.NewStringList(types.DefaultTypeAdapter, result)
 }
+
+func estimateExtractCost() checker.FunctionEstimator {
+	return func(c checker.CostEstimator, target *checker.AstNode, args []checker.AstNode) *checker.CallEstimate {
+		if len(args) == 2 {
+			targetSize := estimateSize(c, args[0])
+			targetCost := targetSize.Add(checker.FixedSizeEstimate(1)).MultiplyByCostFactor(common.StringTraversalCostFactor)
+			regexCost := estimateSize(c, args[1]).Add(checker.FixedSizeEstimate(1)).MultiplyByCostFactor(common.RegexStringLengthCostFactor)
+			return &checker.CallEstimate{CostEstimate: checker.CostEstimate(targetCost).Multiply(regexCost), ResultSize: &targetSize}
+		}
+		return nil
+	}
+}
+
+func estimateExtractAllCost() checker.FunctionEstimator {
+	return func(c checker.CostEstimator, target *checker.AstNode, args []checker.AstNode) *checker.CallEstimate {
+		if len(args) == 2 {
+
+			targetSize := estimateSize(c, args[0])
+			targetCost := targetSize.Add(checker.FixedSizeEstimate(1)).MultiplyByCostFactor(common.StringTraversalCostFactor)
+			regexCost := estimateSize(c, args[1]).Add(checker.FixedSizeEstimate(1)).MultiplyByCostFactor(common.RegexStringLengthCostFactor)
+
+			resultSize := checker.SizeEstimate{Min: 0, Max: targetSize.Max}
+			listAllocCost := resultSize.Add(checker.FixedSizeEstimate(common.ListCreateBaseCost)).MultiplyByCostFactor(1)
+
+			return &checker.CallEstimate{
+				CostEstimate: targetCost.Multiply(regexCost).Add(listAllocCost),
+				ResultSize:   &resultSize,
+			}
+		}
+		return nil
+	}
+}
+
+func estimateReplaceCost() checker.FunctionEstimator {
+	return func(c checker.CostEstimator, target *checker.AstNode, args []checker.AstNode) *checker.CallEstimate {
+		l := len(args)
+		if l == 3 || l == 4 {
+			targetSize := estimateSize(c, args[0])
```

Comment:
I think to be most accurate we need to look at the size of the replacement as well:

```go
targetSize := estimateSIze(c, args[0])
replacementSize := estimateSize(c, args[2])

allReplacedSize := targetSize.Max * replacementSize.Max
noneReplacedSize := targetSize.Max
resultSize := &checker.SizeEstimate{Min: noneReplacedSize, Max: allReplacedSize}
if noneReplacedSize > allReplacedSize {
  resultSize =  &checker.SizeEstimate{Min: allReplacedSize, Max: noneReplacedSize}
}
```